tool.go

```diff
@@ -38,39 +38,39 @@ type ComposeConfig struct {
-	DependsOn           []string               `yaml:"depends_on,omitempty"`
-	DNS                 []string               `yaml:"dns,omitempty"`
-	DNSOpt              []string               `yaml:"dns_opt,omitempty"`
-	DNSSearch           []string               `yaml:"dns_search,omitempty"`
-	DrainTimeoutMs      string                 `yaml:"drain_timeout_ms,omitempty"`
-	Volumes             map[string]interface{} `yaml:"volumes,omitempty"`
-	VolumesFrom         []string               `yaml:"volumes_from,omitempty"`
-	VolumeDriver        string                 `yaml:"volume_driver,omitempty"`
-	Networks            map[string]interface{} `yaml:"networks,omitempty"`
-	NetworkMode         string                 `yaml:"network_mode,omitempty"`
-	Labels              map[string]string      `yaml:"labels,omitempty"`
-	Isolation           string                 `yaml:"isolation,omitempty"`
-	Sysctls             map[string]string      `yaml:"sysctls,omitempty"`
-	Secrets             map[string]interface{} `yaml:"secrets,omitempty"`
-	SecurityOpt         []string               `yaml:"security_opt,omitempty"`
-	StopSignal          string                 `yaml:"stop_signal,omitempty"`
-	Ulimits             map[string]interface{} `yaml:"ulimits,omitempty"`
-	MemLimit            string                 `yaml:"mem_limit,omitempty"`
-	MemReservation      string                 `yaml:"mem_reservation,omitempty"`
-	MemSwapLimit        string                 `yaml:"memswap_limit,omitempty"`
-	MemSwappiness       string                 `yaml:"mem_swappiness,omitempty"`
-	MilliCPUReservation string                 `yaml:"milli_cpu_reservation,omitempty"`
+	Image               string                   `yaml:"image,omitempty"`
+	CgroupParent        string                   `yaml:"cgroup_parent,omitempty"`
+	Links               []string                 `yaml:"links,omitempty"`
+	ExternalLinks       []string                 `yaml:"external_links,omitempty"`
+	Ports               []string                 `yaml:"ports,omitempty"`
+	Expose              []string                 `yaml:"expose,omitempty"`
+	DeviceReadBps       []string                 `yaml:"device_read_bps,omitempty"`
+	DeviceReadIOps      []string                 `yaml:"device_read_iops,omitempty"`
+	Devices             []string                 `yaml:"devices,omitempty"`
+	DeviceWriteBps      []string                 `yaml:"device_write_bps,omitempty"`
+	DeviceWriteIOps     []string                 `yaml:"device_write_iops,omitempty"`
+	DependsOn           []string                 `yaml:"depends_on,omitempty"`
+	DNS                 []string                 `yaml:"dns,omitempty"`
+	DNSOpt              []string                 `yaml:"dns_opt,omitempty"`
+	DNSSearch           []string                 `yaml:"dns_search,omitempty"`
+	DrainTimeoutMs      string                   `yaml:"drain_timeout_ms,omitempty"`
+	Volumes             []map[string]interface{} `yaml:"volumes,omitempty"`
```

Comment:
Volumes should be []string

```
-	Volumes             []map[string]interface{} `yaml:"volumes,omitempty"`
+	Volumes             []string                 `yaml:"volumes,omitempty"`
```

Reply:
I tasted with this docker-compose.yml construct and just using []string does not work. It will lead you to another Unmarshal error.
```
version: '3.2'
services:
  web:
    image: sdelements/lets-chat
    stdin_open: true
    tty: true
    links:
    - database:mongo
    ports:
    - 9890:8080/tcp
    labels:
      io.rancher.container.pull_image: always
      io.rancher.scheduler.global: 'true'
    networks:
      - frontend
      - backend  
    volumes:
      - type: volume
        source: mydata
        target: /data
        volume:
          nocopy: true
      - type: bind
        source: ./static
        target: /opt/app/static
```

Reply:
But this is version 3.2 of compose file. On Rancher v1.x just version 1 and 2 supported..

Reply:
ok, long syntax for 'volumes' is supported since version "2.3", will check about our compatability and change the type